controllers/druid/handler.go

```diff
@@ -391,6 +395,158 @@ func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
 	}
 }
 
+// scalePVCForSts shall scale validate pvc scaling, delete sts using orphan and patch pvc.
+func scalePVCForSts(sdk client.Client, drd *v1alpha1.Druid, ls map[string]string, configMapSHA, serviceName, nodeSpecUniqueStr string, nodeSpec *v1alpha1.DruidNodeSpec) {
+
+	sts := makeStatefulSetEmptyObj()
+	// During creation/initalization of druid cluster, this will fail since the sts will not be present
+	// So logging this as info, event type normal, subsequent reconciles should work fine.
+	if err := sdk.Get(context.TODO(), *namespacedName(nodeSpecUniqueStr, drd.Namespace), sts); err != nil {
+		msg := fmt.Sprintf("failed to get [StatefuleSet:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+		logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+		sendEvent(sdk, drd, v1.EventTypeNormal, "GET_FAIL", err.Error())
+		return
+	}
+
+	// ignoring error, function shall log and sendEvent for it
+	pvc := getPVCList(sdk, nodeSpec, drd, nodeSpecUniqueStr)
+
+	desVolumeClaimTemplateSize, currVolumeClaimTemplateSize, pvcSize := getVolumeClaimTemplateSizes(sts, nodeSpec, pvc)
+
+	ok := checkIfStsHasEverythingElseSame(sdk, drd, ls, configMapSHA, serviceName, nodeSpecUniqueStr, nodeSpec)
+	if !ok {
+		return
+	}
+
+	// Validate Request, shrinking of pvc not supported
+	// desired size cant be less than current size
+	// in that case re-create sts/pvc which is a user execute manual step
+	if desVolumeClaimTemplateSize.String() < currVolumeClaimTemplateSize.String() {
+		msg := fmt.Sprintf("Request for Shrinking of sts pvc size [sts:%s] in [namespace:%s] is not Supported", sts.Name, sts.Namespace)
+		logger.Info(msg)
```

Comment:
this should be `ERROR` as user is not expected to try and downscale the PVC size.